content/specs/did.md

```diff
@@ -76,8 +76,62 @@ did:method:val?two
+A valid signing key is required for atproto functionality, and an account with no valid key in their DID document is broken.
+
+The **PDS service network location** for the account is found under the `service` array, with `id` matching `#atproto_pds`, and `type` matching `AtprotoPersonalDataServer`. The first matching entry in the array should be used, and any others ignored. The `serviceEndpoint` field must contain an HTTPS URL of server. It should contain only the URI scheme (`http` or `https`), hostname, and optional port number, not any "userinfo", path prefix, or other components.
+
+A working PDS is required for atproto functionality, and an account with no valid PDS location in their DID document is broken.
+
+Note that a valid URL doesn't mean the the PDS itself is currently functional or hosting content for the account. During account migrations or server downtime there may be windows when the PDS is not accessible, but this does not mean the account should immediately be considered broken or invalid.
+
+## Public Key Encoding
+
+The atproto cryptographic systems are described in the [AT Protocol Overview](https://www.notion.so/specs/atp).
+
+Public keys in DID documents under `verificationMethod`, including atproto signing keys, are represented as an object with the following fields:
+
+- `id` (string, required): always `#atproto` for atproto signing keys
+- `type` (string, required): a fixed name identifying the key's curve type
+    - `p256`: `EcdsaSecp256r1VerificationKey2019` (note the "r")
+    - `k256`: `EcdsaSecp256k1VerificationKey2019` (note the "k")
+- `controller` (string, required): DID controlling the key, which in the current version of atproto must match the account DID itself
+- `publicKeyMultibase` (string, required): the public key itself, encoded in multibase format
+
+A summary of the multibase encoding in this context:
+
+- Start with the full public key bytes. Do not use the "compressed" or "compact" representation (unlike for `did:key`, described below)
+- Encode the key bytes with `base58btc`, yielding a string
+- Add the character `z` as a prefix, to indicate the multibase, and include no other multicodec indicators
+
+The decoding process is the same in reverse, using the curve type as context.
+
+As an internal detail of the DID PLC method, the W3C-standardized `did:key` encoding is used to represent public keys in DID PLC operations. Software only needs to process keys in this format if they are trying to validate PLC operation chains, but the details are described here anyways.
+
+This encoding includes metadata about the type of key, so they can be parsed and used unambiguously. The encoding process is:
+
+- Encode the public key curve "point" as bytes. Be sure to use the smaller "compact" or "compressed" representation. This is usually easy for `k256`, but might require a special argument or configuration for `p256` keys
+- Prepend the appropriate curve multicode indicator bytes in front of the key bytes:
```

Comment:
this has been a recurring point of confusion in did:key implentations. These bytes do not match the multicodec table for the keys because they are the varint encoded version of those bytes. We just hardcoded the varint encoding

So maybe a quick note that these are the _already varint encoded_ multicodec bytes